geoutils/geovector.py

```diff
@@ -162,15 +165,31 @@ def create_mask(
             transform = rio.transform.from_bounds(left, bottom, right, top, width, height)
 
         # otherwise use directly rst's dimensions
+        elif isinstance(rst, gu.Raster):
+            out_shape = rst.shape
+            transform = rst.transform
+            crs = rst.crs
+            bounds = rst.bounds
         else:
-            out_shape = rst.shape  # type: ignore
-            transform = rst.transform  # type: ignore
-            crs = rst.crs  # type: ignore
+            raise ValueError("`rst` must be either a str, geoutils.Raster or None")
+
+        # Crop vector geometries to avoid issues when reprojecting
+        left, bottom, right, top = bounds  # type: ignore
+        x1, y1, x2, y2 = warp.transform_bounds(crs, self.ds.crs, left, bottom, right, top)
+        self.ds = self.ds.cx[x1:x2, y1:y2]
```

Comment:
This is permanent, so the Vector will be modified after the function call!

Reply:
Good catch! I need to add a copy.

Reply:
Fixed in eb21601